internal/proxy/proxy_test.go

```diff
@@ -0,0 +1,260 @@
+package proxy
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+// MockMultiplexer implements the multiplexer interface for testing
+type MockMultiplexer struct {
+	mock.Mock
+}
```

Comment:
nit: Would prefer to avoid Mock objects if possible, better to have a more explicit minimal implementation for testing purposes.